diff --git a/src/orders/views/OrderReturn/useRefundWithinReturn.ts b/src/orders/views/OrderReturn/useRefundWithinReturn.ts
index 5623bbcb4..d50ddc202 100644
--- a/src/orders/views/OrderReturn/useRefundWithinReturn.ts
+++ b/src/orders/views/OrderReturn/useRefundWithinReturn.ts
@@ -86,7 +86,7 @@ export const squashLines = (items: GrantRefundInputLine[]): GrantRefundInputLine
       (acc, item) => ({
         ...acc,
         [item.id]: acc[item.id]
-          ? { ...item, quantity: acc[item.id].quantity + item.quantity }
+          ? { ...acc[item.id], quantity: acc[item.id].quantity + item.quantity }
           : item,
       }),
       {},
