diff --git a/src/products/components/ProductListDatagrid/datagrid.ts b/src/products/components/ProductListDatagrid/datagrid.ts
index 86093c30b..c299c636e 100644
--- a/src/products/components/ProductListDatagrid/datagrid.ts
+++ b/src/products/components/ProductListDatagrid/datagrid.ts
@@ -85,6 +85,11 @@ export const productListStaticColumnAdapter = ({
       title: intl.formatMessage(columnsMessages.created),
       width: 300,
     },
+    {
+      id: "lastModifiedAt",
+      title: intl.formatMessage(columnsMessages.lastModifiedAt),
+      width: 300,
+    },
     {
       id: "price",
       title: intl.formatMessage(columnsMessages.price),
@@ -221,6 +226,8 @@ export function createGetCellContent({
         return getDateCellContent(rowData);
       case "created":
         return getCreatedCellContent(rowData);
+      case "lastModifiedAt":
+        return getLastModifiedAtCellContent(rowData);
       case "productCategory":
         return getCategoryCellContent(theme, rowData);
       case "productCollections":
@@ -247,6 +254,10 @@ function getCreatedCellContent(rowData: RelayToFlat<ProductListQuery["products"]
   return dateCell(rowData?.created, COMMON_CELL_PROPS);
 }
 
+function getLastModifiedAtCellContent(rowData: RelayToFlat<ProductListQuery["products"]>[number]) {
+  return dateCell(rowData?.updatedAt, COMMON_CELL_PROPS);
+}
+
 function getProductTypeCellContent(
   theme: DefaultTheme,
   rowData: RelayToFlat<ProductListQuery["products"]>[number],
diff --git a/src/products/components/ProductListDatagrid/messages.ts b/src/products/components/ProductListDatagrid/messages.ts
index 356224689..cecee69e1 100644
--- a/src/products/components/ProductListDatagrid/messages.ts
+++ b/src/products/components/ProductListDatagrid/messages.ts
@@ -40,6 +40,11 @@ export const columnsMessages = defineMessages({
     defaultMessage: "Last updated",
     description: "product updated at",
   },
+  lastModifiedAt: {
+    id: "lastModifiedAt",
+    defaultMessage: "Last modified at",
+    description: "product last modified at",
+  },
   created: {
     id: "acgbDa",
     defaultMessage: "Created at",
diff --git a/src/products/urls.ts b/src/products/urls.ts
index e3e4f4e6d..16d549536 100644
--- a/src/products/urls.ts
+++ b/src/products/urls.ts
@@ -75,6 +75,7 @@ export enum ProductListUrlSortField {
   rank = "rank",
   date = "date",
   created = "created",
+  lastModifiedAt = "lastModifiedAt",
 }
 type ProductListUrlSort = Sort<ProductListUrlSortField>;
 export interface ProductListUrlQueryParams
diff --git a/src/products/views/ProductList/sort.ts b/src/products/views/ProductList/sort.ts
index db2fcaa0d..6ea47ccc0 100644
--- a/src/products/views/ProductList/sort.ts
+++ b/src/products/views/ProductList/sort.ts
@@ -11,6 +11,7 @@ export function canBeSorted(sort: ProductListUrlSortField, isChannelSelected: bo
     case ProductListUrlSortField.productType:
     case ProductListUrlSortField.date:
     case ProductListUrlSortField.created:
+    case ProductListUrlSortField.lastModifiedAt:
     case ProductListUrlSortField.attribute:
     case ProductListUrlSortField.rank:
       return true;
@@ -38,6 +39,8 @@ function getSortQueryField(sort: ProductListUrlSortField): ProductOrderField {
       return ProductOrderField.DATE;
     case ProductListUrlSortField.created:
       return ProductOrderField.CREATED_AT;
+    case ProductListUrlSortField.lastModifiedAt:
+      return ProductOrderField.LAST_MODIFIED_AT;
     default:
       return undefined;
   }
